code/datums/dna/dna.dm

```diff
@@ -37,19 +9,21 @@ GLOBAL_LIST_INIT(standard_mutation_sources, list(MUTATION_SOURCE_ACTIVATED, MUTA
 /proc/populate_total_ui_len_by_block()
 	. = list()
 	var/total_block_len = 1
-	for(var/blocknumber in 1 to DNA_UNI_IDENTITY_BLOCKS)
-		. += total_block_len
-		total_block_len += GET_UI_BLOCK_LEN(blocknumber)
+	for(var/block_path as anything in GLOB.dna_identity_blocks)
+		var/datum/dna_block/identity/block = GLOB.dna_identity_blocks[block_path]
+		.[block_path] += total_block_len
+		total_block_len += block.block_length
 
 ///Ditto but for unique features. Used by the datum/dna/set_uni_feature_block and datum/dna/get_uni_feature_block procs.
 GLOBAL_LIST_INIT(total_uf_len_by_block, populate_total_uf_len_by_block())
 
 /proc/populate_total_uf_len_by_block()
 	. = list()
 	var/total_block_len = 1
-	for(var/blocknumber in 1 to DNA_FEATURE_BLOCKS)
-		. += total_block_len
-		total_block_len += GET_UF_BLOCK_LEN(blocknumber)
+	for(var/block_path as anything in GLOB.dna_feature_blocks)
```

Comment:
You don't need `as anything` for untyped vars

Reply:
This went through like 4 iterations and I forgot to change it